src/JSInterop/Microsoft.JSInterop/src/Infrastructure/TaskGenericsUtil.cs

```diff
@@ -89,7 +89,9 @@ public void SetResult(object tcs, object? result)
             // If necessary, attempt a cast
             var typedResult = result is T resultT
                 ? resultT
-                : (T)Convert.ChangeType(result, typeof(T), CultureInfo.InvariantCulture)!;
+                : result == null && default(T) == null // ChangeType can't convert null to value types
```

Comment:
It seems OK to me. 

@oroztocil please review

Reply:
Why `default(T) == null` instead of `typeof(T).IsValueType`?
`IsValueType` is an intrinsic that will be folded by the compiler.

Reply:
> typeof(T).IsGenericType && typeof(T).GetGenericTypeDefinition() == typeof(Nullable<>)

This seems like a better option to me

Reply:
@hez2010 I didn't know about intrinsics and folding, and honestly still don't entirely understand it. I changed it to check for the generic type.

Reply:
> and honestly still don't entirely understand it

The git is able to detect that the expression will be constant for a given T and it's able to optimize it entirely when it generates the machine code for a given T. For example, if you have something like `T = int`, it's able to know that the condition will always evaluate to `false` and it can then use that information to simplify the check (and in this case, unconditionally select one branch or another))